context/config_type.go

```diff
@@ -5,38 +5,35 @@ import (
 	"fmt"
 	"io"
 	"os"
-	"strings"
 
 	"gopkg.in/yaml.v3"
 )
 
 const defaultHostname = "github.com"
 const defaultGitProtocol = "https"
 
-type NotFoundError struct {
-	error
-}
-
+// This interface describes interacting with some persistent configuration for gh.
 type Config interface {
 	Hosts() ([]*HostConfig, error)
-	ConfigForHost(string) (*HostConfig, error)
-	DefaultHostConfig() (*HostConfig, error)
 	Get(string, string) (string, error)
 	Set(string, string, string) error
 	Write() error
+	configForHost(string) (*HostConfig, error)
+	parseHosts(*yaml.Node) ([]*HostConfig, error)
```

Comment:
It's unusual to me to see un-exported methods in an interface. But if it helps the implemenation then I see no harm! Alternatively, you could consider just having package-level functions `configForHost()` and `parseHosts()` that accept the config object as parameter, but then you'd have to have a switch-type statement and that could become messy.

Reply:
it's leftover from when LegacyHost was a thing. That is no longer a thing, so I just removed those from the interface.